float32 escape_angle
float32 escape_distance
bool escape_possible
